pkgs/applications/misc/streamdeck-ui/default.nix

```diff
@@ -0,0 +1,77 @@
+{ lib
+, python3Packages
+, wrapQtAppsHook
+, writeText
+, makeDesktopItem
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "streamdeck-ui";
+  version = "2.0.4";
+
+  src = python3Packages.fetchPypi {
+    pname = "streamdeck_ui";
+    inherit version;
+    sha256 = "650bef0da4957e0d2e2b4cd9c74a55715bcf1526d9ef2cdd04271364117a5845";
+  };
+
+  # Add udev rules to allow non-root access to the Stream Deck.
+  # To enable these rules on NixOS, add this to your system configuration:
+  #
+  #     services.udev.packages = [ pkgs.streamdeck-ui ];
+  postInstall =
+    let
+      udevRulesName = "70-streamdeck.rules";
+      udevRules = ''
+        SUBSYSTEM=="usb", ATTRS{idVendor}=="0fd9", ATTRS{idProduct}=="0060", TAG+="uaccess"
+        SUBSYSTEM=="usb", ATTRS{idVendor}=="0fd9", ATTRS{idProduct}=="0063", TAG+="uaccess"
+        SUBSYSTEM=="usb", ATTRS{idVendor}=="0fd9", ATTRS{idProduct}=="006c", TAG+="uaccess"
+        SUBSYSTEM=="usb", ATTRS{idVendor}=="0fd9", ATTRS{idProduct}=="006d", TAG+="uaccess"
+        SUBSYSTEM=="usb", ATTRS{idVendor}=="0fd9", ATTRS{idProduct}=="0080", TAG+="uaccess"
+      '';
+      desktopItem =
+        (makeDesktopItem {
+          name = "streamdeck-ui";
+          desktopName = "Stream Deck UI";
+          icon = "streamdeck-ui";
+          exec = "streamdeck --no-ui";
+          comment = "UI for the Elgato Stream Deck";
+          categories = [ "Utility" ];
+          noDisplay = true;
+        });
+    in
+      ''
+        mkdir -p "$out/etc/udev/rules.d"
+        cp ${writeText udevRulesName udevRules} $out/etc/udev/rules.d/${udevRulesName}
+
+        mkdir -p "$out/share/applications" "$out/share/pixmaps"
+        cp ${desktopItem}/share/applications/* $out/share/applications
+        cp streamdeck_ui/logo.png $out/share/pixmaps/streamdeck-ui.png
+      '';
+
+  dontWrapQtApps = true;
+  makeWrapperArgs = [ "\${qtWrapperArgs[@]}" ];
+
+  nativeBuildInputs = [ wrapQtAppsHook ];
+
+  propagatedBuildInputs = with python3Packages; [
+    setuptools
+    filetype
+    cairosvg
+    pillow
+    pynput
+    pyside2
+    streamdeck
+    xlib
+  ];
+
+  # Skip tests because they require an X server
```

Comment:
Can we use xvfb-run?

Reply:
I gave that a try, and it solved the missing X errors. But the tests don't seem to be working anyway, so all of the tests had to be skipped. I'm not sure how to debug them further.

Should I push the changes to use `xvfb-run`, even though all of the tests would be skipped? Or would it be better to leave it as-is?